src/Aspire.Hosting.Azure/api/Aspire.Hosting.Azure.cs

```diff
@@ -82,6 +88,26 @@ public static partial class AzureResourceExtensions
         public static ApplicationModel.IResourceBuilder<T> PublishAsConnectionString<T>(this ApplicationModel.IResourceBuilder<T> builder)
             where T : ApplicationModel.IAzureResource, ApplicationModel.IResourceWithConnectionString { throw null; }
     }
+
+    public static partial class ExistingAzureResourceExtensions
+    {
+        public static ApplicationModel.IResourceBuilder<T> AsExisting<T>(this ApplicationModel.IResourceBuilder<T> builder, ApplicationModel.IResourceBuilder<ApplicationModel.ParameterResource> nameParameter, ApplicationModel.IResourceBuilder<ApplicationModel.ParameterResource>? resourceGroupParameter = null)
```

Comment:
@captainsafia - was there a reason this doesn't have `string` overloads like `PublishAsExisting` and `RunAsExisting` have?

Reply:
This is intentionally. Since the `AsExisting` API will run on both run/dev mode scenarios, using a parameter resource serves as a forcing function to ensure that users don't end up targeting the same resource in dev/production modes. What do you think about this?

Reply:
I think that reasoning tracks. But it does feel like a hole just looking at the APIs. An easy workaround is:

```C#
var name = "myAI";
var resourceGroup = "myRG";

var azureAI = builder.AddAzureOpenAI("azureai")
    .RunAsExisting(name, resourceGroup)
    .PublishAsExisting(name, resourceGroup);
```

We can always add this in the future if we decide we need it. So let's not do it now.